node/node_explorer.go

```diff
@@ -148,3 +149,26 @@ func (node *Node) commitBlockForExplorer(block *types.Block) {
 		node.Consensus.PbftLog.DeleteMessagesLessThan(curNum - 100)
 	}
 }
+
+// CommitCommittee commits committee with shard id and epoch to explorer service.
+func (node *Node) CommitCommittee() {
+	blockHeight := uint64(0)
+	for range time.Tick(1000 * time.Millisecond) {
+		curBlock := node.Blockchain().CurrentBlock()
+		curBlockHeight := curBlock.NumberU64()
+		if curBlockHeight == blockHeight {
+			continue
+		}
+		blockHeight = curBlockHeight
+		state, err := node.Blockchain().ReadShardState(curBlock.Epoch())
+		if err != nil {
+			utils.Logger().Error().Err(err).Msg("[Explorer] Error reading shard state")
+			continue
+		}
+		for _, committee := range state {
+			if committee.ShardID == curBlock.ShardID() {
+				explorer.GetStorageInstance(node.SelfPeer.IP, node.SelfPeer.Port, false).DumpCommittee(curBlock.ShardID(), curBlock.Epoch().Uint64(), committee)
+			}
+		}
+	}
+}
```

Comment:
Instead of polling every 1s, please consider using a push-based mechanism.  I remember the blockchain (`core.Blockchain`) providing a hook mechanism for notifications.  If that doesn't work, try locating where `core.(*Blockchain).InsertBlocks` is called and add a hook there instead.